src/App.tsx

```diff
@@ -2,49 +1,83 @@
+import React, { useState } from 'react';
 import 'bulma/css/bulma.css';
-import './App.scss';
+import cn from 'classnames';
 
-export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
+import './App.scss';
+import { SortType } from './types/SortType';
+import { goodsFromServer } from './mocks';
+import { getPreparedGoods } from './utils/getPreparedGoods';
 
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | ''>('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const handleClear = () => {
+    setSortField('');
+    setIsReversed(false);
+  };
+
+  const isChanged = sortField !== '' || isReversed;
+
+  const goods = getPreparedGoods(goodsFromServer, { sortField, isReversed });
+
+  const buttons = [
+    {
+      name: 'Sort alphabetically',
+      className: cn('is-info', { 'is-light': sortField !== SortType.Alphabet }),
+      handleClick() {
+        setSortField(SortType.Alphabet);
+      },
+    },
+    {
+      name: 'Sort by length',
+      className: cn('is-success', {
+        'is-light': sortField !== SortType.Length,
+      }),
+      handleClick() {
+        setSortField(SortType.Length);
+      },
+    },
+    {
+      name: 'Reverse',
+      className: cn('is-warning', { 'is-light': !isReversed }),
+      handleClick() {
+        setIsReversed(prevState => !prevState);
```

Comment:
The function name should start with a verb to clearly indicate the action it performs. Consider renaming this function to follow this convention, perhaps `handleReverse`. Refer to the checklist item 4 about function naming.